src/set.rs

```diff
@@ -1580,9 +1588,9 @@ mod tests {
         assert_eq!(&set_a & &set_b, empty);
         assert_eq!(&set_b & &set_a, empty);
         assert_eq!(&set_a | &set_b, set_c);
-        assert_eq!(&set_b | &set_a, set_c);
+        assert_ne!(&set_b | &set_a, set_c);
         assert_eq!(&set_a ^ &set_b, set_c);
-        assert_eq!(&set_b ^ &set_a, set_c);
+        assert_ne!(&set_b ^ &set_a, set_c);
```

Comment:
when this is revived later, this change does not preserve the intent of the test. It can of course be fleshed out when/if relevant later.